// @sap-cloud-sdk/core/dist/odata-v4/uri-conversion/odata-uri.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.oDataUriV4 = void 0;
var odata_common_1 = require("../../odata-common");
var get_expand_1 = require("./get-expand");
var get_select_1 = require("./get-select");
var uri_value_converter_1 = require("./uri-value-converter");
var getFilter = odata_common_1.createGetFilter(uri_value_converter_1.uriConverterV4).getFilter;
var getResourcePathForKeys = odata_common_1.createGetResourcePathForKeys(uri_value_converter_1.uriConverterV4).getResourcePathForKeys;
var convertToUriFormat = uri_value_converter_1.uriConverterV4.convertToUriFormat;
/**
 * Instance of the [[ODataUri]] conversion interface for OData v4.
 */
exports.oDataUriV4 = {
    getExpand: function (_, expands, entityConstructor) {
        return get_expand_1.getExpandV4(expands, entityConstructor);
    },
    getFilter: getFilter,
    getEntityKeys: odata_common_1.getEntityKeys,
    getOrderBy: odata_common_1.getOrderBy,
    getResourcePathForKeys: getResourcePathForKeys,
    getSelect: get_select_1.getSelectV4,
    convertToUriFormat: convertToUriFormat
};
